### PR TITLE
Frb refactor

### PR DESCRIPTION
**Pull request by profxj**
Fixes #  .

## Description of the Change

Adds new FRB functionality to FFFF-PZ.  This includes
the `FRBTransient`, `FRBGalaxy`, and `Path` models.

These are exposed in view and the API.

## Checklist

Please check all that apply to your proposed changes

- [ X] HTML code change
- [ ] Added package dependency
- [ ] Added data
- [ X] Changed django model
- [ ] Documentation change

### CODE REVIEW

**Review by lordrick94 — APPROVED** [review]
Amazing work!!

File: YSE_App/models/transient_models.py [hunk 1]
```diff
@@ -21,329 +21,359 @@
...
-			return None
-
-	def __str__(self):
-		return self.name
-
-	def natural_key(self):
-		return self.name
+    has_hst = models.BooleanField(null=True, blank=True)
+    has_spitzer = models.BooleanField(null=True, blank=True)
+    has_chandra = models.BooleanField(null=True, blank=True)
```
Comment (tarraneh) [inline YSE_App/models/transient_models.py:93]: Also check for `has_vlass`? e.g., via CIRADA?

File: YSE_App/models/transient_models.py [hunk 2]
```diff
@@ -21,329 +21,359 @@
...
-				thachertag = TransientTag.objects.get(name='Thacher')
-				instance.tags.add(thachertag)
-			except: pass
-			
-		instance.save()
-		#if is_k2_C19_validated:
-		#	coord_string = GetSexigesimalString(instance.ra, instance.dec)
-		#	coord_string = instance.CoordString()
-		#	SendTransientAlert(instance.id, instance.name, coord_string[0], coord_string[1])
+    tag_K2 = False
```
Comment (tarraneh) [inline YSE_App/models/transient_models.py:297]: Define

File: YSE_App/models/transient_models.py [hunk 3]
```diff
@@ -21,329 +21,359 @@
...
+
+    if created:
+        print("Transient Created: %s" % instance.name)
+        print("Internal Survey: %s" % instance.internal_survey)
+
+        if tag_K2:
```
Comment (tarraneh) [inline YSE_App/models/transient_models.py:303]: Define difference in CXX values

File: YSE_App/templates/YSE_App/dashboard.html [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 {% block content_header %}
     <h1>
-        CHIME/FRB Outriggers
+        YSE/PZ
```
Comment (tarraneh) [inline YSE_App/templates/YSE_App/dashboard.html:10]: Why YSE/PZ and not CHIME/FRB Outriggers?

File: YSE_App/tests/test_chime.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    import path
+except ModuleNotFoundError:
+    from YSE_App.galaxies import path
+
+from IPython import embed
+
+def internal_test_of_path():
+    """ Check PATH analysis internally (i.e. not with the DB)"""
+    # Load up the table
+    chime_frbs = pandas.read_csv('../chime/chime_tests.csv')
```
Comment (tarraneh) [inline YSE_App/tests/test_chime.py:18]: Error `FileNotFoundError: [Errno 2] No such file or directory: '../chime/chime_tests.csv'` when running from shell

File: YSE_App/chime/chime_path_test.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+from YSE_App.galaxies import path
+
+from YSE_App.serializers import frbtransient_serializers 
+from YSE_App.serializers import user_serializers
+
+import pandas
+
+from IPython import embed
+
+def run(delete_existing:bool=True,
```
[the remark this answers is not in the document]
Comment (profxj) [reply YSE_App/chime/chime_path_test.py:28]: the next PR will include a `repeater` attribute
in the FRB Transient model.  But we are going
to rely on the CHIME team to indicate if it is a repeater.

For Outrigger localizations, this should be easy (he says..)

[the remark this answers is not in the document]
Comment (profxj) [reply YSE_App/chime/chime_path_test.py:28]: fixed!

File: YSE_App/chime/chime_path_test.py [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    if 'CHIME/FRB' not in survey_names:
+        obs = FRBSurvey(name='CHIME/FRB', created_by_id=user.id, modified_by_id=user.id)
+        obs.save()
+
+    # Add em (if necessary)
+    _ = ctu.add_df_to_db(df_frbs, user, 
+                         delete_existing=delete_existing)
+
+    # Transient for PATH
+    itransient = FRBTransient.objects.get(name='FRB20300714A')
```
[the remark this answers is not in the document]
Comment (profxj) [reply YSE_App/chime/chime_path_test.py:62]: yes, good suggestion

save for Target PR

File: YSE_App/chime/chime_tags_test.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    # Add em
+    dbtransients = ctu.add_df_to_db(df_frbs, user, 
+                                    delete_existing=delete_existing)
+
+    # Test them!
+    for ss in range(len(df_frbs)):
+        ifrb = df_frbs.iloc[ss]
+        assert ifrb['name'] in [t.name for t in Transient.objects.all()]
+        # Tag
+        if ifrb['name'] == 'FRB20300102B':
```
[the remark this answers is not in the document]
Comment (profxj) [reply YSE_App/chime/chime_tags_test.py:50]: good catch;  changed `Transient` -> `FRBTransient`

File: YSE_App/chime/tags.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
+""" This module will specify the tags
```
[the remark this answers is not in the document]
Comment (profxj) [reply YSE_App/chime/tags.py:1]: agreed;  also for future PR

File: YSE_App/data_utils.py [hunk 10]
```diff
@@ -1422,3 +1427,170 @@ def getRADecBox(ra,dec,size=None,dec_size=None):
...
+     in its data (all in JSON, of course; 
+     data types are for after parsing the JSON):
+
+      - transient_name (str): Name of the FRBTransient object
+        Must be in the DB already
+      - table (str): a table of the PATH candidates and their 
+        PATH results, stored as JSON
+      - F (str): name of filter; must be present in the
+        PhotometricBand table
+      - instrument (str): name of the instrumen; must be present in the
```
[the remark this answers is not in the document]
Comment (profxj) [reply YSE_App/data_utils.py:1554]: fixed

File: YSE_App/galaxies/path.py [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+    represent the cumulative 99.9% poseterior, including P(U|x) with 
+    a maximum of `nmax` candidates.
+
+    Args:
+        instance (): Transient or pandas.Series instance of a Transient
+        ssize (float, optional): Radius for probing the survey (arcmin). Defaults to 5.
+        priors (dict, optional): PATH priors. Defaults to None, in which 
+            case the CHIME defaults are used.
+        min_loc_err (float, optional): _description_. Defaults to 0.2.
+        tot_posterior (float, optional): Total posterior probability to
```
[the remark this answers is not in the document]
Comment (profxj) [reply YSE_App/galaxies/path.py:153]: edited

File: YSE_App/models/transient_models.py (hunk 1 above)
Comment (profxj) [reply YSE_App/models/transient_models.py:93]: good ideas;  would have to implement

File: YSE_App/models/transient_models.py (hunk 2 above)
Comment (profxj) [reply YSE_App/models/transient_models.py:297]: reverting all of this

File: YSE_App/templates/YSE_App/dashboard.html (hunk 4 above)
Comment (profxj) [reply YSE_App/templates/YSE_App/dashboard.html:10]: we use `frb_dashboard.html`

this is the original (and I reverted it back)

File: YSE_App/tests/test_chime.py (hunk 5 above)
Comment (profxj) [reply YSE_App/tests/test_chime.py:18]: this is deprecated;  am removing the file

**Review by tarraneh — APPROVED** [review]

**Pull request merged**